### PR TITLE
MultiChain gumby experiments

### PR DESCRIPTION
**Pull request by snorberhuis**
I will squash the changes when approved.

### CODE REVIEW

File: experiments/multichain/multichain_client.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    def __init__(self, *args, **kwargs):
+        super(MultiChainDelayCommunity, self).__init__(*args, **kwargs)
+
+    def allow_signature_request(self, message):
+        """
+        Ignore the signature requests.
+        :param message: the to be delayed request
+        """
+        self.logger.info("Received signature request that will delayed for %s." % self.delay)
+        sleep(self.delay)
```
Comment (whirm) [inline experiments/multichain/multichain_client.py]: no sleeps

File: experiments/dispersy/hiddenservices_client.py [hunk 2]
```diff
@@ -147,9 +150,9 @@ def cb(ds):
             download.set_state_callback(cb, delay=1)
 
             # Force lookup
-            sleep(10)
-            logging.error("Do a manual dht lookup call to bootstrap it a bit")
-            self._community.do_dht_lookup(tdef.get_infohash())
+            # sleep(10)
+            # msg("Do a manual dht lookup call to bootstrap it a bit")
+            # self._community.do_dht_lookup(tdef.get_infohash())
```
Comment (whirm) [inline experiments/dispersy/hiddenservices_client.py]: Why has this been commented out?

File: experiments/multichain/multichain_client.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        """
+        Ignore the signature requests.
+        :param message: the to be ignored request
+        """
+        self.logger.info("Received signature request that will be ignored.")
+        return
+
+if __name__ == '__main__':
+    MultiChainClient.scenario_file = environ.get('SCENARIO_FILE', 'multichain_standalone.scenario')
+    main(MultiChainClient)
```
Comment (whirm) [inline experiments/multichain/multichain_client.py]: missing end of the line. I suggest you do an autopep8 pass to all the files you have created.

File: experiments/multichain/multichain_crawler.conf [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+#
+# Number of sync clients we should wait for to be registered before starting the experiment. (default is DAS4_INSTANCES_TO_RUN)
+sync_subscribers_amount = 1
+
+use_local_venv = FALSE
+
+local_output_dir ="output"
+
+# Dir where to write all the output generated from the experiment (default is workspace_dir/output)
+ output_dir = "output/multichain"
```
Comment (whirm) [inline experiments/multichain/multichain_crawler.conf]: missing newline

File: experiments/multichain/multichain_integrated.conf [hunk 5]
```diff
@@ -0,0 +1,454 @@
+# This is a gumby experiment config file
+# Description experiment: Simple gumby experiment to learn gumby.
```
Comment (whirm) [inline experiments/multichain/multichain_integrated.conf]: I guess it isn't :)

File: experiments/multichain/multichain_integrated.conf [hunk 6]
```diff
@@ -0,0 +1,454 @@
...
+# head_nodes = node1,node2,node3
+#
+
+tracker_cmd = 'run_tracker.sh'
+# tracker_run_remote =
+tracker_port = 7788
+
+experiment_server_cmd = 'experiment_server.py'
+
+#
```
Comment (whirm) [inline experiments/multichain/multichain_integrated.conf]: Remove all the commented out stuff that's not relevant to the experiment.

File: experiments/multichain/processing/DatabaseReader.py [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+from Tribler.community.multichain.community import GENESIS_ID
+from Tribler.community.multichain.conversion import PK_LENGTH
+
+EMPTY_HASH_ENCODED = base64.encodestring(EMPTY_HASH)
+GENESIS_ID_ENCODED = base64.encodestring(GENESIS_ID)
+
+
+class MultiChainExperimentAnalysisDatabase(MultiChainDB):
+    """
+    Extended MultiChain experiments that provides additional analysis functionality to analysis a experiment.
```
Comment (whirm) [inline experiments/multichain/processing/DatabaseReader.py]: *additional functionality to analyze an experiment

**Comment by whirm** [comment]
I'll have a closer look later. 
Could I get links to all the experiments on jenkins?
Thanks!

File: experiments/multichain/multichain_client.py (hunk 1 above)
Comment (snorberhuis) [inline experiments/multichain/multichain_client.py]: How can I create a delay then on purpose?
This community is for testing how MultiChain copes with late requests.

Comment (whirm) [inline experiments/multichain/multichain_client.py]: have a look at reactor.callLater for instance. I suggest reading a twisted intro to get an idea of what can be done.

**Comment by whirm** [comment]
Closing as the work is continuing on #240

**Pull request closed without merging**